hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -334,26 +333,6 @@ def test_function(self, data):
 
             if self.shrinks >= MAX_SHRINKS:
                 self.exit_with(ExitReason.max_shrinks)
-        if (
-            self.settings.timeout > 0
-            and benchmark_time() >= self.start_time + self.settings.timeout
-        ):
-            note_deprecation(
-                (
-                    "Your tests are hitting the settings timeout (%.2fs). "
-                    "This functionality will go away in a future release "
-                    "and you should not rely on it. Instead, try setting "
-                    "max_examples to be some value lower than %d (the number "
-                    "of examples your test successfully ran here). Or, if you "
-                    "would prefer your tests to run to completion, regardless "
-                    "of how long they take, you can set the timeout value to "
-                    "hypothesis.unlimited."
-                )
-                % (self.settings.timeout, self.valid_examples),
-                since="2018-07-29",
```

Comment:
That's true, but the timeout setting itself has been deprecated for significantly longer - and in v4 it's an error to set it to a number rather than `unlimited`.  This deprecation warning is therefore dead code, and could only be retained if we also kept the `timeout` setting (to which: noooo :scream:).